src/client/sandbox/event/drag-and-drop/data-transfer.js

```diff
@@ -0,0 +1,314 @@
+// NOTE: Although DataTransfer interface has a constructor, it is not possible to
+// create a useful DataTransfer object from script, since DataTransfer objects have a
+// processing and security model that is coordinated by the browser during drag-and-drops.
+// So we have to create a mock for it to use it in drag-and-drop events
+import { setTimeout } from '../../native-methods';
+
+const DROP_EFFECT = {
+    none: 'none',
+    copy: 'copy',
+    link: 'link',
+    move: 'move'
+};
+
+const EFFECT_ALLOWED = {
+    uninitialized: 'uninitialized',
+    none:          'none',
+    copy:          'copy',
+    copyLink:      'copyLink',
+    copyMove:      'copyMove',
+    link:          'link',
+    linkMove:      'linkMove',
+    move:          'move',
+    all:           'all'
+};
+
+const DATA_TRANSFER_ITEM_KIND = {
+    string: 'string',
+    file:   'file'
+};
+
+const DATA_STORE_MODE = {
+    readwrite: 'readwrite',
+    readonly:  'readonly',
+    protected: 'protected'
+};
+
+function parseTextUriList (textUriList) {
+    textUriList = textUriList.replace(/\r\n$/, '');
+
+    var res = [];
+
+    if (textUriList === '')
+        return res;
+
+    textUriList = textUriList.split(/\r\n/);
+
+    for (var i = 0; i < textUriList.length; i++) {
+        if (textUriList[0] !== '#')
+            res.push(textUriList[i]);
+    }
+
+    return res;
+}
+
+
+class DataStore {
+    constructor () {
+        this.mode = DATA_STORE_MODE.readwrite;
+    }
+}
+
+// https://html.spec.whatwg.org/multipage/interaction.html#datatransferitem
+class DataTransferItem {
```

Comment:
I thought about that. In the one hand this classes are closely related with each other, on the other hand they can be separated without any problems. So If you propose it, I'll split them